addons/bestja_detailed_reports/models.py

```diff
@@ -171,6 +193,27 @@ def _compute_project_dates(self):
         return "{} — {}".format(date_start, date_stop)
 
     @api.one
+    @api.depends('project', 'report_entries', 'project.organization.level', 'project.children.stores')
+    def _compute_total_cities_nr(self):
+        """
+        Computes number of unique cities where stores
+        are held, both for the organization and all its children.
+        For children of the organization it is set to 0, so that it can
+        be used in the graph view.
+        """
+        project = self.sudo().project
+        level = project.organization.level
+        if level == 1:
+            cities_total = set()
+            for store in project.children.stores:
+                cities_total.add(store.store.city)
+            for store in project.stores:
+                cities_total.add(store.store.city)
+            self.total_cities_nr = len(cities_total)
```

Comment:
I guess that simple `self.total_cities_nr = len(set(project.children.stores) | set(project.stores))` won’t work?